### ISSUE TITLE
Found with fuzzer: break_wand -> dig_check -> is_lightsaber

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#39]
```

Program received signal SIGSEGV, Segmentation fault.
0x00005555558b2145 in dig_check (madeby=0x0, verbose=0 '\000', x=44, y=16) at dig.c:193
193	              ? "dig in" : is_lightsaber(uwep) 
#0  0x00005555558b2145 in dig_check (madeby=0x0, verbose=0 '\000', x=44, y=16) at dig.c:193
#1  0x00005555557ba7be in do_break_wand (obj=0x60c000ecf8c0) at apply.c:4589
#2  0x00005555557bc4e9 in doapply () at apply.c:4796
#3  0x0000555555866bf4 in rhack (cmd=0x5555567bd300 <in_line> "a") at cmd.c:5531
#4  0x0000555555766f7b in moveloop (resuming=0 '\000') at allmain.c:823
#5  0x00005555563ce396 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  0x00005555558b2145 in dig_check (madeby=0x0, verbose=0 '\000', x=44, y=16) at dig.c:193
        ttmp = 0x0
        verb = 0x7fffffffe3e0 ""
#1  0x00005555557ba7be in do_break_wand (obj=0x60c000ecf8c0) at apply.c:4589
        typ = 0 '\000'
        nothing_else_happens = "But nothing else happens..."
        i = 0
        x = 44
        y = 16
        mon = <optimized out>
        dmg = 20
        damage = 0
        affects_objects = 0 '\000'
        shop_damage = 0 '\000'
        fillmsg = 0 '\000'
        expltype = 4
        confirm = <optimized out>
        buf = <optimized out>
        is_fragile = 0 '\000'
#2  0x00005555557bc4e9 in doapply () at apply.c:4796
        obj = 0x60c000ecf8c0
        otmp = 0x0
        res = 1
        class_list = "\006\002\v\000\000\000\000\000@\305FVUU\000\000\357\017\207UU"
#3  0x0000555555866bf4 in rhack (cmd=0x5555567bd300 <in_line> "a") at cmd.c:5531
        tlist = 0x55555667a380 <extcmdlist+192>
        res = 688
        func = 0x5555557bbfbb <doapply>
        spkey = 0
        prefix_seen = 0 '\000'
        bad_command = -87 '\251'
        firsttime = 1 '\001'
#4  0x0000555555766f7b in moveloop (resuming=0 '\000') at allmain.c:823
        moveamt = 12
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 26337
        past_clock = 86634
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#5  0x00005555563ce396 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
This is caused by applying/breaking a wand of digging when we don't have any weapons wielded. There is a check to see if a lightsaber is wielded and it references null uwep.

**Comment by elunna** [comment]
Fixed in commit  e19179d75:  dig_check needed to guard against no weapon being wielded when breaking a wand of digging.
